aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -93,6 +93,107 @@ func resourceAwsSagemakerEndpointConfiguration() *schema.Resource {
 			},
 
 			"tags": tagsSchema(),
+
+			"data_capture_config": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enable_capture": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"initial_sampling_percentage": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.IntBetween(0, 100),
+						},
+
+						"destination_url": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validateSagemakerDataCaptureDestinationUrl,
+						},
+
+						"kms_key_id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validateArn,
+						},
+
+						"capture_options": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 2,
+							MinItems: 1,
+							ForceNew: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"capture_mode": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ForceNew:     true,
+										ValidateFunc: validation.StringInSlice([]string{"Input", "Output"}, false),
+									},
+								},
+							},
+						},
+
+						"capture_content_type_header": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							ForceNew: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"csv_content_types": {
+										Type:     schema.TypeList,
```

Comment:
I think we can similify this to look like this:

```
"csv_content_types": {
	Type:     schema.TypeSet,
	MinItems: 1,
	MaxItems: 10,
    Elem:     &schema.Schema{
      Type: schema.TypeString,
      ValidateFunc: validateSagemakerCsvContentType,                                         },
	Optional: true,
	ForceNew: true,
},
```

this will also look more similar to the API as it corresponds to slices `CsvContentTypes []*string`